core-java-modules/core-java-11-2/src/test/java/com/baeldung/httpclient/ssl/HttpClientSSLBypassUnitTest.java

```diff
@@ -4,29 +4,84 @@
 import org.junit.jupiter.api.Test;
 
 import java.io.IOException;
+import java.net.Socket;
 import java.net.URI;
+import java.net.URISyntaxException;
 import java.net.http.HttpClient;
 import java.net.http.HttpRequest;
 import java.net.http.HttpResponse;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.X509Certificate;
 import java.util.Properties;
 
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509ExtendedTrustManager;
+
 public class HttpClientSSLBypassUnitTest {
 
     @Test
-    public void whenHttpsRequest_thenCorrect() throws IOException, InterruptedException {
+    public void whenHttpsRequest_withDisabledHostName_thenCorrect() throws IOException, InterruptedException {
         final Properties props = System.getProperties();
         props.setProperty("jdk.internal.httpclient.disableHostnameVerification", Boolean.TRUE.toString());
 
         HttpClient httpClient = HttpClient.newBuilder()
-          .build();
+            .build();
 
         HttpRequest request = HttpRequest.newBuilder()
-          .uri(URI.create("https://wrong.host.badssl.com/"))
-          .build();
+            .uri(URI.create("https://wrong.host.badssl.com/"))
+            .build();
 
         HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
         props.setProperty("jdk.internal.httpclient.disableHostnameVerification", Boolean.FALSE.toString());
 
         Assertions.assertEquals(200, response.statusCode());
     }
+
+    @Test
+    public void whenHttpsRequest_withMOCK_CERT_thenCorrect() throws IOException, InterruptedException, NoSuchAlgorithmException, KeyManagementException, URISyntaxException {
```

Comment:
can we rename this to, givenMockTrustManager_whenByPassCertificateVerification_thenCorrect()